lib/internal/mime.js

```diff
@@ -0,0 +1,318 @@
+'use strict';
+
+const {
+  ArrayPrototypeJoin,
+  ObjectCreate,
+  MapIteratorPrototypeNext,
+  RegExpPrototypeTest,
+  RegExpPrototypeExec,
+  StringPrototypeReplace,
+  StringPrototypeToLowerCase,
+  StringPrototypeSlice,
+  StringPrototypeCharAt,
+  SymbolReplace,
+  SymbolIterator,
+  SafeMap: Map,
+} = primordials;
+
+const NotHTTPTokenCodePoint = /[^!#$%&'*+\-.^_`|~A-Za-z0-9]/g;
+const NotHTTPQuotedStringCodePoint = /[^\t\u0020-~\u0080-\u00FF]/g;
+NotHTTPQuotedStringCodePoint[SymbolReplace] = null;
+
+const HTTP_TAB_OR_SPACE = '\t ';
+const HTTP_WHITESPACE = ArrayPrototypeJoin(['\r\n', ...HTTP_TAB_OR_SPACE], '');
+const BEGINNING_WHITESPACE = new RegExp(
+  `^[${StringPrototypeReplace(HTTP_WHITESPACE, /\W/g, '\\$&')}]*`);
+const ENDING_WHITESPACE = new RegExp(
+  `[${StringPrototypeReplace(HTTP_WHITESPACE, /\W/g, '\\$&')}]*$`);
+
+function toASCIILower(str) {
+  return StringPrototypeReplace(str, /[A-Z]/g, (c) => StringPrototypeToLowerCase(c));
+}
+
+function parseTypeAndSubtype(str) {
+  // Skip only HTTP whitespace from start
+  str = StringPrototypeReplace(str, BEGINNING_WHITESPACE, '');
+  let position = 0;
+  // read until '/'
+  const typeMatch = execPattern(/^[^/]*/, str);
```

Comment:
Same here, `const slashIndex = str.indexOf('/')` + `str.slice(0, slashIndex)`?

Reply:
see above / -1 results make it a bit juggly